### PR TITLE
Add Kindling Emitter

### PR DESCRIPTION
**Pull request by akarahdev**
Currently, Fire has no easy way of emitting Kindling besides just generating a string. However, I made a group of classes that aim to drastically simplify this process - allowing you to generate Kotlin classes that can be emitted into valid Kindling. It's not fully done, as I'm unsure if the Kindling it emits is valid.

Input:
```kt
val toEmit = KindlingHeader.PlayerEvent("Join", arrayOf(
    KindlingAction.PlayerAction("SendMessage", "", arrayOf(KindlingArgument.Text("hi %default"))),
))

val output = emitValue(emitHeader(toEmit))
println(output)
```
Output:
```
( player-event "Join" ( ( player-action "SendMessage" "" ( ( text "hi %default" ) ) ) ) )
```

### CODE REVIEW

**Comment by akarahdev** [comment]
Wait, I forgot to finish the actions. I'll get to that in a bit.

File: src/main/kotlin/dev/ashli/fire/parser/emitter/KindlingEmitter.kt
Comment (ashlikatt) [inline src/main/kotlin/dev/ashli/fire/parser/emitter/KindlingEmitter.kt:1]: I would like to see this broken up into different files with JavaDocs to explain their use, even if trivial.

**Review by ashlikatt — CHANGES_REQUESTED** [review]
Looks mostly good, just would like to see better organization and documentation. This is sure to help a ton once we actually get to a point where we're generating Kindling!

**Comment by akarahdev** [comment]
Alright, thanks for the input! I finished the actions, fixed some issues, reorganized, and added JavaDocs for all of it.

**Review by ashlikatt — APPROVED** [review]
Much better, love it!

**Pull request merged**